helpers.lua

```diff
@@ -889,8 +889,12 @@ function logout()
         flash("info", t("logged_out"))
     end
 
-    -- Redirect to portal anyway
-    return redirect(conf.portal_url)
+    -- Redirect with the `r` URI argument if it exists or redirect to portal
+    if args.r then
```

Comment:
I think it should be in the "if is_logged_in() then" No ?

Reply:
No, I don't think so. The function `logout()` already assumes that the user is logged in (as per line 884) and delete their cookies/session. Then my proposed code adds a check for the `r` argument to use it for redirection, and otherwise defaults to the initial redirection to portal.